frontend/src/components/pages/GlobalForms/WaiverAddSectionModal.tsx

```diff
@@ -0,0 +1,100 @@
+import {
+  Box,
+  Button,
+  Checkbox,
+  HStack,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  Text,
+  Textarea,
+} from "@chakra-ui/react";
+import React from "react";
+import { WaiverClause } from "../../../types/AdminTypes";
+
+type WaiverAddSectionModalProps = {
+  isOpen: boolean;
+  onClose: () => void;
+  onAddWaiverSectionClick: (newClause: WaiverClause) => void;
+};
+
+const WaiverAddSectionModal = ({
+  isOpen,
+  onClose,
+  onAddWaiverSectionClick,
+}: WaiverAddSectionModalProps): JSX.Element => {
+  const [newClauseText, setNewClauseText] = React.useState("");
+  const [clauseRequired, setClauseRequired] = React.useState(false);
+
+  const handleInputChange = (e: any) => {
+    const inputValue = e.target.value;
+    setNewClauseText(inputValue);
+  };
+
+  const handleCheckboxChange = () => {
+    setClauseRequired(!clauseRequired);
+  };
+
+  const submitModal = () => {
+    const newClause: WaiverClause = {
+      text: newClauseText,
+      required: clauseRequired,
+    };
+    onAddWaiverSectionClick(newClause);
+    setNewClauseText("");
+    setClauseRequired(false);
+    onClose();
+  };
+
+  return (
+    <Modal isOpen={isOpen} onClose={onClose} isCentered preserveScrollBarGap>
+      <ModalOverlay />
+      <ModalContent minWidth="30vw">
+        <ModalHeader textStyle="heading">
+          <Text>Add a new waiver section</Text>
+        </ModalHeader>
+        <ModalCloseButton />
+        <ModalBody>
+          <Box justifyContent="space-evenly">
+            <Text textStyle="bodyRegular">Waiver section</Text>
+            <Textarea
```

Comment:
can you set resizable to false but make it scrollable? Right now, the user can resize the text area which makes the modal look really weird